src/main/java/hudson/remoting/Checksum.java

```diff
@@ -54,4 +60,38 @@ public int hashCode() {
     public String toString() {
         return String.format("%016X%016X",sum1,sum2);
     }
+
+    /**
+     * Returns the checksum for the given file.
+     */
+    static Checksum forFile(File file) throws IOException {
+        return forURL(file.toURI().toURL());
+    }
+
+    /**
+     * Returns the checksum for the given URL.
+     */
+    static Checksum forURL(URL url) throws IOException {
+        try {
+            MessageDigest md = MessageDigest.getInstance(JarLoaderImpl.DIGEST_ALGORITHM);
+            Util.copy(url.openStream(), new DigestOutputStream(new NullOutputStream(), md));
+            return new Checksum(md.digest(), md.getDigestLength() / 8);
+        } catch (NoSuchAlgorithmException e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    private static class NullOutputStream extends OutputStream {
```

Comment:
Unfortunately neither of them are :(.

Guava is there but only for tests. I feel like it should be added as a dependency for remoting, I've come across a few places where having it would have been nice. It would increase the slave.jar size, but I haven't looked into by how much.

I'd prefer to leave adding Guava to a separate pull request if its decided to be included. Let me know if it is - I'll be happy to bring it in.